app/script/media/MediaStreamHandler.js

```diff
@@ -545,53 +568,50 @@ z.media.MediaStreamHandler = class MediaStreamHandler {
    * Check for active calls that need a MediaStream.
    * @returns {boolean} Returns true if an active media stream is needed for at least one call
    */
-  needs_media_stream() {
-    for (const call_et of this.calls()) {
-      const is_incoming_video_call =
-        call_et.isRemoteVideoSend() && call_et.state() === z.calling.enum.CALL_STATE.INCOMING;
+  needsMediaStream() {
+    for (const callEntity of this.calls()) {
+      const isIncomingCall = callEntity.state() === z.calling.enum.CALL_STATE.INCOMING;
+      const isIncomingVideoCall = isIncomingCall && callEntity.isRemoteVideoSend();
 
-      if (call_et.selfClientJoined() || is_incoming_video_call) {
+      if (callEntity.selfClientJoined() || isIncomingVideoCall) {
         return true;
       }
     }
 
     return false;
   }
 
-  // Toggle the camera.
-  toggle_video_send() {
-    const hasActiveVideoStream = this.localMediaStream() && this.local_media_type() === z.media.MediaType.VIDEO;
-    return hasActiveVideoStream ? this._toggle_video_send() : this.replace_input_source(z.media.MediaType.VIDEO);
-  }
-
   // Toggle the mute state of the microphone.
-  toggle_audio_send() {
-    const hasActiveAudioStream = !!this.localMediaStream();
-    return hasActiveAudioStream
-      ? this._toggleAudioSend()
-      : Promise.reject(new z.media.MediaError(z.media.MediaError.TYPE.NO_AUDIO_STREAM_FOUND));
+  toggleAudioSend() {
+    return this._toggleAudioSend();
   }
 
   // Toggle the screen.
-  toggle_screen_send() {
-    const hasActiveScreenStream = this.localMediaStream() && this.local_media_type() === z.media.MediaType.SCREEN;
-    return hasActiveScreenStream ? this._toggleScreenSend() : this.replace_input_source(z.media.MediaType.SCREEN);
+  toggleScreenSend() {
+    const hasActiveScreenStream = this.localMediaStream() && this.localMediaType() === z.media.MediaType.SCREEN;
+    return hasActiveScreenStream ? this._toggleScreenSend() : this.replaceInputSource(z.media.MediaType.SCREEN);
```

Comment:
I see this piece of code duplicated twice.
We could refactor this in another private function that takes a type and a toggle function parameter

```
toggleMediaState(type, toggleFn) {
  const hasActiveScreenStream = this.localMediaStream() && this.localMediaType() === type;
  return hasActiveScreenStream ? toggleFn() : this.replaceInputSource(type);
}
```

and then in each function

```
toggleScreenSend() {
  return this.toggleMediaState(SCREEN, this._toggleScreenSend.bind(this));
}
```

Reply:
Why exactly do we check the state of the mediaStream in the boolean toggling function?
Since this check is already (partially) done in the `_toggleStreamEnabled`, that should be enough, isn't it?

Reply:
If it's a matter of exposing a state that depends on both the internal state AND the real state of the mediaStream, then we should create a new observable that depends on both.
Something like

```
isAudioSent = ko.pureComputed(() => {
  return this.state.audioSend() && this.localMediaStream(AUDIO).enabled;
});
```